gptel-transient.el

```diff
@@ -185,22 +185,18 @@ which see."
   (transient-parse-suffixes
    'gptel-system-prompt
    (cl-loop for (type . prompt) in gptel-directives
-       with taken
+       ;; Avoid clashes with the custom directive key
+       with unused-keys = (remove ?h (number-sequence ?a ?z))
        with width = (window-width)
        for name = (symbol-name type)
-       for key =
-       (let ((idx 0) pos)
-         (while (or (not pos) (member pos taken))
-           (setq pos (substring name idx (1+ idx)))
-           (cl-incf idx))
-         (push pos taken)
-         pos)
+       for key = (seq-find (lambda (k) (member k unused-keys)) name (seq-first unused-keys))
+       do (setq unused-keys (delete key unused-keys))
```

Comment:
This is going to cons up a ~26-element sequence as many times as there are directives, creating a bunch of garbage.  It's not much but it adds up (with all the other allocation gptel does), and this is why I chose to not do it this way.

Reply:
The doc says that `delete` works in-place like `delq` but you are supposed to reassign the result value anyway to make clear what happens, so that is what I did. I can also change it to `delq` if you prefer.

Reply:
Just checked the implementation of `delete`, I think we're good here -- it appears to do additional consing if `unused-keys` is a vector, not a list.